app/controllers/admin/users_controller.rb

```diff
@@ -0,0 +1,48 @@
+class Admin::UsersController < Admin::ApplicationController
+  def show
+    @user = User.find(params[:id])
+  end
+
+  def index
+    @q = User.ransack(params[:q])
+    @users = @q.result.order(:type).order(:id).page(params[:page]).per(10)
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      redirect_to(admin_users_url)
+    else
+      render(:new_admin_user_url)
+    end
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update(user_params)
+      redirect_to(admin_users_url)
+    else
+      render(:edit)
+    end
+  end
+
+  def destroy
+    @user = User.find(params[:id])
+    @user.destroy
+    redirect_to(admin_users_path)
```

Comment:
An empty line before the final or return statement would make your code easier to read. Keep it in mind in the future.